iaso/api/org_units.py

```diff
@@ -503,6 +527,20 @@ def partial_update(self, request, pk=None):
         else:
             return Response(errors, status=400)
 
+    @staticmethod
+    def check_date_format(date, format):
+        try:
+            return bool(datetime.strptime(date, format))
+        except ValueError:
+            return False
+
+    @staticmethod
+    def get_date(self, date):
+        if self.check_date_format(date, "%d-%m-%Y"):
+            return datetime.strptime(date, "%d-%m-%Y").date()
+        else:
+            return datetime.strptime(date, "%d/%m/%Y").date()
```

Comment:
This is a bit hard to read  because you have to jump between multiple functions:

- read the code calling `self.get_date()`
- then go read `get_date()`
- then go read `check_date_format()`

Here is a one function alternative (if you like it):

```python
def get_date(self, date: str) -> typing.Union[datetime.date, False]:
    date_input_formats = ["%d-%m-%Y", "%d/%m/%Y"]
    for date_input_format in date_input_formats:
        try:
            return datetime.strptime(date, date_input_format).date()
        except ValueError:
            pass
    return False
```

Also, as a `@staticmethod` this could be unit tested in isolation.